src/gstrtpreceiver.cpp

```diff
@@ -153,14 +162,35 @@ static void loop_pull_appsink_samples(bool& keep_looping,GstElement *app_sink_el
 std::string GstRtpReceiver::construct_gstreamer_pipeline()
 {
     std::stringstream ss;
-    if (! unix_socket)
-        ss<<"udpsrc port="<<m_port<<" "<<pipeline::gst_create_rtp_caps(m_video_codec)<<" ! ";
-    else
-        ss<<"appsrc name=appsrc "<<pipeline::gst_create_rtp_caps(m_video_codec)<<" ! ";
-    ss<<pipeline::create_rtp_depacketize_for_codec(m_video_codec);
-    ss<<pipeline::create_parse_for_codec(m_video_codec);
-    ss<<pipeline::create_out_caps(m_video_codec);
-    ss<<" appsink drop=true name=out_appsink";
+
+    // Branch 1: Primary UDP/appsrc stream
+    if (!unix_socket) {
+        ss << "udpsrc name=udpsrc port=" << m_port << " "
+           << pipeline::gst_create_rtp_caps(m_video_codec) << " ! "
+           << pipeline::create_rtp_depacketize_for_codec(m_video_codec)
+           << pipeline::create_parse_for_codec(m_video_codec)
+           << "selector.sink_0 ";
+    } else {
+        ss << "appsrc name=appsrc " << pipeline::gst_create_rtp_caps(m_video_codec) << " ! "
+           << pipeline::create_rtp_depacketize_for_codec(m_video_codec)
+           << pipeline::create_parse_for_codec(m_video_codec)
+           << "selector.sink_0 ";
+    }
+
+    // Banch 2: Backup filesrc
+    if (splash_enabled) {
+    ss << "filesrc name=filesrc location=\"" <<  splash_file  << "\" ! qtdemux ! "
+        << "queue max-size-buffers=0 max-size-time=0 max-size-bytes=0 ! "
+        << pipeline::create_parse_for_codec(m_video_codec)
+        << "selector.sink_1 ";
```

Comment:
if I start a DVR while splash is playing, would it record the splash?

Also, if I lost the video link with the drone (it went out of range) would it mean the splash going to start?